packages/plugins/graphql/server/services/builders/filters/content-type.js

```diff
@@ -61,6 +75,31 @@ module.exports = ({ strapi }) => {
     });
   };
 
+  const addComponentScalarAttribute = (builder, attributeName, contentType, attribute) => {
```

Comment:
I'm not sure to understand the need to only handle scalar component attributes here?

A simpler implementation of a addComponentAttribute might look like that

```js
const addComponentAttribute = (builder, attributeName, attribute) => {
    const utils = strapi.plugin('graphql').service('utils');
    const extension = strapi.plugin('graphql').service('extension');
    const { getFiltersInputTypeName } = utils.naming;

    const component = strapi.getModel(attribute.component);

    // If there is no component corresponding to the attribute configuration, then ignore it
    if (!component) return;

    // If the component is disabled, then ignore it too
    if (extension.shadowCRUD(component.uid).isDisabled()) return;

    builder.field(attributeName, { type: getFiltersInputTypeName(component) });
}
```

Reply:
A small explanation for the code snippet above:

Here we're only defining an input type for the filters of a given content type. Thus we don't need to add resolvers or args as you did below. We just need to indicate the GraphQL type (let's call it `T`) of a component attribute in the filters.

This type `T` is already defined by calling the [registerFiltersDefinition](https://github.com/strapi/strapi/blob/graphql-nested-component-filters/packages/plugins/graphql/server/services/content-api/index.js#L157) for each content type **AND every component** in the [registerAPITypes](https://github.com/strapi/strapi/blob/graphql-nested-component-filters/packages/plugins/graphql/server/services/content-api/index.js#L147) method of the content-api service.

With that in mind, we can safely assume that there is a filter type (`T`) definition for every component and that we can reuse those types for our component attributes.

This is why instead of dealing with resolver logic or even defining new types here, all we have to do is to check if the component field is enabled and -_if it's not_- add its filter type (T`)`) to the parent type.